remoulade/common.py

```diff
@@ -25,7 +25,38 @@
 from .errors import QueueJoinTimeout
 
 
-def compute_backoff(attempts, *, factor=5, jitter=True, max_backoff=2000, max_exponent=32):
+def compute_backoff(attempts, *, strategy=None, factor=5, jitter=True, max_backoff=2000, max_exponent=32):
+    """Compute the backoff using the selected strategy.
+
+    Parameters:
+      attempts(int): The number of attempts there have been so far.
+      strategy(str): The selected strategy.
+      factor(int): The number of milliseconds to multiply each backoff by.
+      jitter(bool): If true adds a small random value to the backoff to avoid mass simultaneous retries.
+      max_backoff(int): The max number of milliseconds to backoff by.
+      max_exponent(int): The maximum backoff exponent.
+    """
+
+    if strategy == "spread_exponential":
+        backoff = compute_backoff_spread_exponential(
+            attempts, factor=factor, max_backoff=max_backoff, max_exponent=max_exponent
+        )
+    elif strategy == "constant":
+        backoff = factor
+    elif strategy == "linear":
+        backoff = compute_backoff_linear(attempts, factor=factor, max_backoff=max_backoff)
+    else:
+        backoff = compute_backoff_exponential(
+            attempts, factor=factor, max_backoff=max_backoff, max_exponent=max_exponent
+        )
+    if jitter:
+        backoff *= 3 / 4
+        backoff = int(backoff + uniform(0, backoff) / 3)
```

Comment:
I would do something like this in case we want to change the factor:
```
jitter_factor = 1/4
backoff = int(backoff * (1 - jitter_factor) + uniform(0, backoff) * jitter_factor)
```